localstack-core/localstack/services/cloudformation/engine/v2/change_set_model_preproc.py

```diff
@@ -630,8 +630,12 @@ def _compute_fn_or(args: list[bool]):
     def visit_node_intrinsic_function_fn_not(
         self, node_intrinsic_function: NodeIntrinsicFunction
     ) -> PreprocEntityDelta:
-        def _compute_fn_not(arg: bool) -> bool:
-            return not arg
+        def _compute_fn_not(arg: list[bool] | bool) -> bool:
+            # Is the argument ever a lone boolean?
```

Comment:
[the docs](http://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-not) define the argument of a Fn::Not is an array. But let's keep the try/except just in case.